nvtabular/io/parquet.py

```diff
@@ -111,8 +111,41 @@ def _dataset(self):
         return dataset
 
     @property
-    @functools.lru_cache(1)
     def _file_partition_map(self):
+        if not hasattr(self, "_pp_map"):
```

Comment:
I prefer attributes to be defined in the constructor. Can we set _pp_map/_pp_nrows to None in the ```__init__``` and then just check if None here (rather than the hasatter check)?